src/ProductConstructionService/ProductConstructionService.Api/Queue/JobProcessors/CodeFlowJobProcessor.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.DotNet.DarcLib;
+using Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+using Microsoft.DotNet.Maestro.Client.Models;
+using ProductConstructionService.Api.Queue.Jobs;
+
+namespace ProductConstructionService.Api.Queue.JobProcessors;
+
+public class CodeFlowJobProcessor(
+        IVmrInfo vmrInfo,
+        IBasicBarClient barClient,
+        IVmrBackFlower vmrBackFlower,
+        IVmrForwardFlower vmrForwardFlower,
+        ILogger<CodeFlowJobProcessor> logger)
+    : IJobProcessor
+{
+    private readonly IVmrInfo _vmrInfo = vmrInfo;
+    private readonly IBasicBarClient _barClient = barClient;
+    private readonly IVmrBackFlower _vmrBackFlower = vmrBackFlower;
+    private readonly IVmrForwardFlower _vmrForwardFlower = vmrForwardFlower;
+    private readonly ILogger<CodeFlowJobProcessor> _logger = logger;
+
+    public async Task ProcessJobAsync(Job job, CancellationToken cancellationToken)
+    {
+        var codeflowJob = (CodeFlowJob)job;
+
+        Subscription subscription = await _barClient.GetSubscriptionAsync(codeflowJob.SubscriptionId)
+            ?? throw new Exception($"Subscription {codeflowJob.SubscriptionId} not found");
+
+        if (!subscription.SourceEnabled || subscription.SourceDirectory == null)
+        {
+            throw new Exception($"Subscription {codeflowJob.SubscriptionId} is not source enabled or source directory is not set");
+        }
+
+        Build build = await _barClient.GetBuildAsync(codeflowJob.BuildId)
+            ?? throw new Exception($"Build {codeflowJob.BuildId} not found");
+
+        var isForwardFlow = build.GitHubRepository != _vmrInfo.VmrUri && build.AzureDevOpsRepository != _vmrInfo.VmrUri;
```

Comment:
Can we add some parentheses for better readability?

Reply:
Sure